pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java

```diff
@@ -75,22 +82,40 @@ public Dictionary getDictionary() {
 
   @Override
   public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
-    throw new UnsupportedOperationException();
+    if (_intResult == null) {
+      _intResult = new int[DocIdSetPlanNode.MAX_DOC_PER_CALL];
+      Arrays.fill(_intResult, Double.valueOf(_literal).intValue());
```

Comment:
Can we follow the same convention as in the `ArrayCopyUtils` so that the behavior is consistent everywhere? (Use Integer.parseInt, Long.parseLong, Float.parseFloat, Double.parseDouble, BytesUtils.toBytes)
If the String is not of the target value format, the current behavior is to throw exception.
For this PR, I don't think you need to change this class because the binary operator first check the data type and always fetch String value from Literal. Ideally it should check for Literal and call `getLiteral()`

Reply:
Reverted the change on this.